src/vs/workbench/contrib/terminalContrib/suggest/browser/terminal.suggest.contribution.ts

```diff
@@ -138,29 +70,68 @@ class TerminalSuggestContribution extends DisposableStore implements ITerminalCo
 			return;
 		}
 		this.add(Event.runAndSubscribe(this._ctx.instance.onDidChangeShellType, async () => {
-			this._loadSuggestAddon(xterm.raw);
+			this._loadAddons(xterm.raw);
 		}));
 		this.add(this._contextKeyService.onDidChangeContext(e => {
 			if (e.affectsSome(this._terminalSuggestWidgetContextKeys)) {
-				this._loadSuggestAddon(xterm.raw);
+				this._loadAddons(xterm.raw);
 			}
 		}));
 		this.add(this._configurationService.onDidChangeConfiguration(e => {
 			if (e.affectsConfiguration(TerminalSettingId.SendKeybindingsToShell)) {
-				this._loadSuggestAddon(xterm.raw);
+				this._loadAddons(xterm.raw);
 			}
 		}));
 	}
 
-	private _loadSuggestAddon(xterm: RawXtermTerminal): void {
+	private _loadPwshCompletionAddon(xterm: RawXtermTerminal): void {
+		if (this._ctx.instance.shellType !== GeneralShellType.PowerShell) {
+			return;
+		}
+		const pwshCompletionProviderAddon = this._pwshAddon.value = this._instantiationService.createInstance(PwshCompletionProviderAddon, undefined, this._ctx.instance.capabilities);
+		xterm.loadAddon(pwshCompletionProviderAddon);
+		this.add(pwshCompletionProviderAddon);
```

Comment:
it's an addon because of https://github.com/microsoft/vscode/blob/10d669fada7adea1e810a31e5d144aa6e9ba7c32/src/vs/workbench/contrib/terminalContrib/suggest/browser/pwshCompletionProviderAddon.ts#L122-L135